protected/humhub/widgets/Pjax.php

```diff
@@ -1,60 +1,19 @@
 <?php
 
-/**
- * @link https://www.humhub.org/
- * @copyright Copyright (c) 2016 HumHub GmbH & Co. KG
- * @license https://www.humhub.com/licences
- */
-
 namespace humhub\widgets;
 
-use Yii;
-use yii\helpers\Json;
-use humhub\assets\PjaxAsset;
+use yii\helpers\ArrayHelper;
 
-/**
- * Pjax Widget
- *
- * @author Luke
- */
-class Pjax extends \humhub\components\Widget
+class Pjax extends \yii\widgets\Pjax
 {
+    public $timeout = 30000;
 
-    /**
-     * @var array options passed to pjax scrpit
-     */
-    public $clientOptions = [];
-
-    /**
-     * @inheritdoc
-     */
     public function init()
     {
-        $this->clientOptions['pushRedirect'] = true;
-        $this->clientOptions['replaceRedirect'] = true;
-        $this->clientOptions['cache'] = false;
-        $this->clientOptions['timeout'] = 5000;
+        $this->options = ArrayHelper::merge($this->options, [
+            'class' => ['exclude-from-pjax-client'],
+        ]);
```

Comment:
Should that not be something like

```php
        $this->options['class'] = 'exclude-from-pjax-client ' . $this->options['class'] ?? '';
```

In order not to overwrite classes specified in the `Pjax::widget/begin` call?

Reply:
@martin-rueegg used `ArrayHelper::merge`, it will merge classes from options

Reply:
@gevorgmansuryan well, I guess the situation is more complex:

- `ArrayHelper::merge()` will indeed merge them if the classes are provided as an array with individual elements.
- However, if `$this->options['class']` is a string, then the the value from the second array will overwrite the string.

I do think this should be considered and merged accordingly. My solution does address the later case, but not the former.